TaskEngine/Tasks/TranscriptionTask.cs

```diff
@@ -65,43 +89,27 @@ protected async override Task OnConsume(string videoId, TaskParameters taskParam
                     });
                 }
             }
-            using (var _context = CTDbContext.CreateDbContext())
-            {
-                var latestVideo = await _context.Videos.FindAsync(video.Id);
-                if (result.Item2 == "NoError")
-                {
-                    if (latestVideo.TranscriptionStatus != "NoError")
-                    {
-                        // If any present, remove them.
-                        if (latestVideo.Transcriptions.Any())
-                        {
-                            var oldTranscriptions = latestVideo.Transcriptions;
-                            var oldCaptions = latestVideo.Transcriptions.SelectMany(t => t.Captions);
-                            _context.Captions.RemoveRange(oldCaptions);
-                            await _context.SaveChangesAsync();
-                            _context.Transcriptions.RemoveRange(oldTranscriptions);
-                            await _context.SaveChangesAsync();
-                        }
-
-                        // Add the latest transcriptions.
-                        await _context.Transcriptions.AddRangeAsync(transcriptions);
-                        await _context.SaveChangesAsync();
-                        transcriptions.ForEach(t => _generateVTTFileTask.Publish(t.Id));
-                        _sceneDetectionTask.Publish(video.Id);
-                    }
-                    latestVideo.TranscriptionStatus = result.Item2;
-                    latestVideo.TranscribingAttempts += 1;
-                    await _context.SaveChangesAsync();
 
-                }
-                else
-                {
-                    latestVideo.TranscriptionStatus = result.Item2;
-                    latestVideo.TranscribingAttempts += 1;
-                    await _context.SaveChangesAsync();
-                    throw new Exception("Transcription failed" + result.Item2);
-                }
+            var latestVideo = await _context.Videos.FindAsync(video.Id);
```

Comment:
Do we need to different variables for video and latestVideo?

Reply:
Changed